### PR TITLE
visual components for RBF

### LINKED ISSUE #3289: Review send modal: Updated
**Design:** [Zeplin](https://zpl.io/anjMRvr) 

This update includes: 

- Left side with TX "parameters", right side with inputs to be signed for more visual clarity
- Success screen with link to TX detail  in case `Broadcast = ON, i.e. signing + broadcast`
- Success screen with buttons to download raw TX in case `Broadcast = OFF, i.e. signing only` 
- New, less technical TX detail view 
- Progress counter in case of 10+ items (?) to be signed by the device

### PR DESCRIPTION
**Pull request by tomexx**
Review send + RBF modal design updated

### CODE REVIEW

File: packages/suite/src/utils/wallet/transactionUtils.ts [hunk 1]
```diff
@@ -312,7 +312,7 @@ export const getRbfParams = (
     account: Account,
 ): RbfTransactionParams | undefined => {
     if (account.networkType !== 'bitcoin') return;
-    if (tx.type === 'recv' || !tx.rbf || !tx.details || !isPending(tx)) return; // ignore non rbf and mined transactions
+    if (tx.type === 'recv' || !tx.rbf || !tx.details /*|| !isPending(tx)*/) return; // ignore non rbf and mined transactions
```
Comment (szymonlesisz) [inline packages/suite/src/utils/wallet/transactionUtils.ts]: bring that back pls

Comment (slowbackspace) [inline packages/suite/src/utils/wallet/transactionUtils.ts]: just scrolled through and noticed it, don't foget to remove the comment!

File: packages/suite/src/components/suite/modals/CoinmarketReviewTransaction/components/Detail.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import React, { useState } from 'react';
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/modals/CoinmarketReviewTransaction/components/Detail.tsx:1]: is this the same as `modals/ReviewTransaction/Details`? could it be reused?

File: packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    hasExpansion = false,
+    fiatVisible = false,
+}: Props) => {
+    const TruncateWrapper = ({
+        condition,
+        children,
+    }: {
+        condition: boolean;
+        children: React.ReactNode;
+    }): any => {
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx]: rm any

File: packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                        <>
+                            <OutputValueWrapper>
+                                <FormattedCryptoAmount
+                                    disableHiddenPlaceholder
+                                    value={value}
+                                    symbol={cryptoSymbol}
+                                />
+                            </OutputValueWrapper>
+                            {fiatVisible && (
+                                <>
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx]: unnecessary fragment?

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/components/suite/modals/CoinmarketReviewTransaction/components/Detail.tsx (hunk 2 above)
Comment (tomexx) [reply packages/suite/src/components/suite/modals/CoinmarketReviewTransaction/components/Detail.tsx:1]: Coinmarket referenced Detail.tsx from ReviewTransaction. As there were changes in Detail.tsx I decided to copy the previous version of it and move it to Coinmarket as its submodule. We can make it reusable for both usages in a next PR

File: packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx (hunk 4 above)
Comment (tomexx) [reply packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx]: Actually it has to be there b/c there are multiple nodes underneath :/

Comment (szymonlesisz) [reply packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx]: damn, i mean the one in line 104

Comment (tomexx) [reply packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx]: I've pushed a more elegant way to do this without fighting with types (thx @slowbackspace)

**Comment by slowbackspace** [comment]
Screenshot 2021-02-02 at 16 55 22
We should probably round the value just like in tx detail modal. @szymonlesisz is that a proper way to go?

Waait I am sending TROP, but fee is calculated in sat/B. 
One option would be to just skip this info on non-btc-like coins, or in case of eth add gas limit/price fields and in case of xrp add their thing.

other issues:
- incorrect icon for RBF/broadcast
- incorrect bg color for detail panel (on the right side)
- fade out looks weird in dark mode
- hard to click expand arrow (additional padding needed)
- missing success part of the modal (after tx is sent)

**Comment by slowbackspace** [comment]
One more problem with the design itself rather than implementation.
We cant fit 18 decimals in this narrow panel. Even putting symbol on new line is more like workaround than actual fix as it would still break on higher amounts.

Created an issue for this https://github.com/trezor/trezor-suite/issues/3323
I don't believe it should block merging this PR, but it should be fixed before the release 🙇 
image

**Comment by tomexx** [comment]
> One more problem with the design itself rather than implementation.
> We cant fit 18 decimals in this narrow panel. Even putting symbol on new line is more like workaround than actual fix as it would still break on higher amounts.
> image

https://github.com/trezor/trezor-suite/pull/3303#issuecomment-771740436

**Comment by tomexx** [comment]
> Screenshot 2021-02-02 at 16 55 22
> 
> We should probably round the value just like in tx detail modal. @szymonlesisz is that a proper way to go?
> Waait I am sending TROP, but fee is calculated in sat/B.
> One option would be to just skip this info on non-btc-like coins, or in case of eth add gas limit/price fields and in case of xrp add their thing.

I'll amend it so it is showing the right thing for the right network, I missed this one

File: packages/suite/src/components/suite/modals/CoinmarketReviewTransaction/components/Detail.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+const Top = styled.div`
+    width: 100%;
+    border-bottom: solid 1px ${props => props.theme.STROKE_GREY};
+    display: flex;
+    flex-direction: row;
+`;
+
+const Label = styled.span`
+    color: #808080;
```
Comment (slowbackspace) [inline packages/suite/src/components/suite/modals/CoinmarketReviewTransaction/components/Detail.tsx]: missing props.theme.TYPE_DARK_GREY?

**Comment by slowbackspace** [comment]
one more visual nitpick, "details" should stick to a bottom
implementation:
Screenshot 2021-02-02 at 18 25 59

design:
Screenshot 2021-02-02 at 18 26 52

**Comment by szymonlesisz** [comment]
RBF confirmation design inconsistencies, there is a missing "double-lined" fee confirmation (it was in layout prototype tho):
Screenshot from 2021-02-02 20-14-48

**Comment by szymonlesisz** [comment]
rbf modal fixed 👍

**Comment by tomexx** [comment]
squashed

**Pull request merged**